### DISCUSSION TITLE
Class Attribute in Tabs Macro

### DISCUSSION THREAD

**Discussion by kookma** [TiddlyWiki/TiddlyWiki5#6147] [Q&A]
* goto http://tiddlywiki.com/
* create a tiddler with below text
```
<<tabs tabsList:"One Two" class:"cl1 cl2">>
```
The in preview pan select raw html, it produces

```
<p>
<div class="tc-tab-set cl1 cl2">
<div class="tc-tab-buttons cl1 cl2">
<button class="">
One
</button><button class="">
Two
</button>
</div>
<div class="tc-tab-divider cl1 cl2"></div>
<div class="tc-tab-content cl1 cl2">
<div class=" tc-reveal" hidden="true"></div><div class=" tc-reveal" hidden="true"></div>
</div>
</div>
</p>
```

Does it mean to have custom class `cl1`, `cl2` repeated for all above html tags? See for example line three and four

```
<div class="tc-tab-set cl1 cl2">
<div class="tc-tab-buttons cl1 cl2">
```

Isn't it better to have different classes here?

**Comment by pmario** [comment]
Most code of the tabs macro is about 8+ years old. see: https://github.com/Jermolene/TiddlyWiki5/blame/d3522854b6e6b3bfdcfed9c6407ad85f1a717f8b/core/wiki/macros/tabs.tid#L18  

The `class` parameter as it is has been there for a long time. 

IMO it depends on what you want to achieve, if the current "setting" makes sense or not. ... It is a possibility to create a user specific styling, but the CSS has to be "very specific", since the TW core defines many elements in a much more specific way as it is convenient for user stylesheets.

**Reply by kookma** [reply]
Thank you @pmario - I believe in the current design and using  class in such a way like beliw

```
<div class="tc-tab-set cl1 cl2">
<div class="tc-tab-buttons cl1 cl2">
```

does not make sense!

In TW-Lab I am working on tabs and have shared part of the results on Talk!
I would be happy to contribute to improve tabs in the core! 
I believe the current design can be simplified and more user friendly for customizing!

**Reply by Jermolene** [reply]
The idea is that by specifying a custom class such as "cl1" that you can then independently target the components of the tab with selectors `.tc-tab-set.cl1` and `.tc-tab-buttons.cl1`. It's the mechanism we use to apply the classes for vertical tabs.
